packages/svelte/src/compiler/compile/render_ssr/index.js

```diff
@@ -144,9 +144,13 @@ export default function ssr(component, options) {
 		? b`
 			let $$settled;
 			let $$rendered;
+			let #previous_head = $$result.head;
 
 			do {
 				$$settled = true;
+				// $$result.head is mutated by the literal expression
+				// need to reset it if we're looping back to prevent duplication
+				$$result.head = #previous_head;
```

Comment:
Still don't fully understand the fix - how is the binding mutating the head?

Reply:
Binding doesn't mutate it, but the emitted code is like
```
`${($$result.head += 'head contents', '')} body contents`
```
so if the loop isn't settled in one go, the next iteration will add the head contents again and so on

Reply:
Oh I see - I guess it makes sense then to only use the first head. Can there ever be a situation where this would prevent other head elements from showing up? Like, if there's some component inside an if block that contains a head element, will that be part of the first loop already?

Reply:
Should be fine, the child component inside the if will get reexecuted after resetting the head

Reply:
So what constitutes to a rerun then? Only bindings? TBH I never looked into that part of the code base much. I just want to rule out that we don't lose head content that would otherwise be rightfully added.

Reply:
https://github.com/sveltejs/svelte/blob/adea6039969f7e8498ac1fb8480df48bdf1b0242/packages/svelte/src/compiler/compile/render_ssr/index.js#L143

Looks like it's only bindings